In [1]:
import os
import glob
import pandas as pd
import matplotlib as plt

In [2]:
base_dir = '/cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/v2_v2.5f0752a_conf2_annotFrame_cap0_runs_pya0.17.1/TCGA_Breast_1102/cohort_mutNone/tmp_out_ref_batch_70'

In [4]:
glob.glob(base_dir + '/*')

['/cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/v2_v2.5f0752a_conf2_annotFrame_cap0_runs_pya0.17.1/TCGA_Breast_1102/cohort_mutNone/tmp_out_ref_batch_70/ref_sample_peptides_meta.pq',
 '/cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/v2_v2.5f0752a_conf2_annotFrame_cap0_runs_pya0.17.1/TCGA_Breast_1102/cohort_mutNone/tmp_out_ref_batch_70/ref_annot_peptides.fa.pq',
 '/cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/v2_v2.5f0752a_conf2_annotFrame_cap0_runs_pya0.17.1/TCGA_Breast_1102/cohort_mutNone/tmp_out_ref_batch_70/output_sample_IS_SUCCESS',
 '/cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/v2_v2.5f0752a_conf2_annotFrame_cap0_runs_pya0.17.1/TCGA_Breast_1102/cohort_mutNone/tmp_out_ref_batch_70/ref_graph_kmer_SegmExpr.pq',
 '/cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/v2_v2.5f0752a_conf2_annotFrame_cap0_runs_pya0.17.1/TCGA_Breast_1102/cohort_mutNone/tmp_out_ref_batch_70/tmp_ref_graph_kmer_Jun

In [6]:
ohsu = pd.read_csv(os.path.join(base_dir, 
            'ENSG00000188157.15_jx_chr1.1053978.1054447.+_info.tsv'), sep = '\t')

In [7]:
eth = pd.read_csv(os.path.join(base_dir, 
            'tmp_ref_graph_kmer_JuncExpr.tsv'), sep = ',')

In [8]:
eth.head()

kmer  isCrossJunction  TCGA-3C-AAAU-01A-11R-A41B-07.all  \
0  MAGRSHPGP            False                               NaN   
1  AGRSHPGPL            False                               NaN   
2  GRSHPGPLR            False                               NaN   
3  RSHPGPLRP            False                               NaN   
4  SHPGPLRPL            False                               NaN   

   TCGA-3C-AALI-01A-11R-A41B-07.all  TCGA-3C-AALJ-01A-31R-A41B-07.all  \
0                               NaN                               NaN   
1                               NaN                               NaN   
2                               NaN                               NaN   
3                               NaN                               NaN   
4                               NaN                               NaN   

   TCGA-3C-AALK-01A-11R-A41B-07.all  TCGA-4H-AAAK-01A-12R-A41B-07.all  \
0                               NaN                               NaN   
1                               NaN                               NaN   
2                               NaN                               NaN   
3                               NaN                               NaN   
4                               NaN                               NaN   

   TCGA-5L-AAT0-01A-12R-A41B-07.all  TCGA-5L-AAT1-01A-12R-A41B-07.all  \
0                               NaN                               NaN   
1                               NaN                               NaN   
2                               NaN                               NaN   
3                               NaN                               NaN   
4                               NaN                               NaN   

   TCGA-5T-A9QA-01A-11R-A41B-07.all  ...  TCGA-UL-AAZ6-01A-11R-A41B-07.all  \
0                               NaN  ...                               NaN   
1                               NaN  ...                               NaN   
2                               NaN  ...                               NaN   
3                               NaN  ...                               NaN   
4                               NaN  ...                               NaN   

   TCGA-UU-A93S-01A-21R-A41B-07.all  TCGA-V7-A7HQ-01A-11R-A33J-07.all  \
0                               NaN                               NaN   
1                               NaN                               NaN   
2                               NaN                               NaN   
3                               NaN                               NaN   
4                               NaN                               NaN   

   TCGA-W8-A86G-01A-21R-A36F-07.all  TCGA-WT-AB41-01A-11R-A41B-07.all  \
0                               NaN                               NaN   
1                               NaN                               NaN   
2                               NaN                               NaN   
3                               NaN                               NaN   
4                               NaN                               NaN   

   TCGA-WT-AB44-01A-11R-A41B-07.all  TCGA-XX-A899-01A-11R-A36F-07.all  \
0                               NaN                               NaN   
1                               NaN                               NaN   
2                               NaN                               NaN   
3                               NaN                               NaN   
4                               NaN                               NaN   

   TCGA-XX-A89A-01A-11R-A36F-07.all  TCGA-Z7-A8R5-01A-42R-A41B-07.all  \
0                               NaN                               NaN   
1                               NaN                               NaN   
2                               NaN                               NaN   
3                               NaN                               NaN   
4                               NaN                               NaN   

   TCGA-Z7-A8R6-01A-11R-A41B-07.all  
0                        

In [9]:
ohsu.head()

jx  TCGA-24-1431-01A-01R-1566-13  \
0  chr1;1053978;1054447;+                   2451.671033   

   TCGA-24-2298-01A-01R-1569-13  TCGA-25-1313-01A-01R-1565-13  \
0                   1623.400763                   1051.181948   

   TCGA-25-1319-01A-01R-1565-13  TCGA-61-2008-01A-02R-1568-13  \
0                    1073.47063                   1526.614818   

   TCGA-A2-A0D2-01A-21R-A034-07  TCGA-A2-A0SX-01A-12R-A084-07  \
0                    246.158054                    452.656363   

   TCGA-AO-A0JM-01A-21R-A056-07  TCGA-BH-A18V-01A-11R-A12D-07  ...  \
0                    651.739725                    707.655859  ...   

   hanging_txs_included_inframe_pepseqs  prefiltered_in-frame_epitopes  \
0                                   NaN                            NaN   

   prefiltered_inframe_epitope_count  in-frame neoepitopes  \
0                                  0                   NaN   

   in-frame_neoepitope_count  \
0                          0   

               frame-agnostic_all-transcript_biexons  \
0  EKALQSNHFELSLRTEATQGLVLWSGKATERADYVALAIVDGHLQL...   

   nonhanging_prefiltered_epitopes  nonhanging_prefiltered_ep_count  \
0                              NaN                              0.0   

   nonhanging_neoepitopes  nonhanging_neoep_count  
0                     NaN                     0.0  

[1 rows x 56 columns]

In [30]:
target_samples = [col for col in eth.columns if ''.join(col.split('.')[:-1]) in ohsu.columns]
target_samples_ohsu = [''.join(col.split('.')[:-1]) for col in target_samples ]

In [20]:

eth_small = eth.loc[:, ['kmer', 'isCrossJunction'] + target_samples ]

In [22]:
eth_small.head()

kmer  isCrossJunction  TCGA-A2-A0D2-01A-21R-A034-07.all  \
0  MAGRSHPGP            False                               NaN   
1  AGRSHPGPL            False                               NaN   
2  GRSHPGPLR            False                               NaN   
3  RSHPGPLRP            False                               NaN   
4  SHPGPLRPL            False                               NaN   

   TCGA-A2-A0SX-01A-12R-A084-07.all  TCGA-AO-A0JM-01A-21R-A056-07.all  \
0                               NaN                               NaN   
1                               NaN                               NaN   
2                               NaN                               NaN   
3                               NaN                               NaN   
4                               NaN                               NaN   

   TCGA-BH-A18V-01A-11R-A12D-07.all  TCGA-C8-A12P-01A-11R-A115-07.all  
0                               NaN                               NaN  
1                               NaN                               NaN  
2                               NaN                               NaN  
3                               NaN                               NaN  
4                               NaN                               NaN

2 possible junction kmers which pass the filtering: SWHIGSRGK, GCGSWHIGS
peptide: PWCCVPPCPSTPTAGCGSWHIGSRGKVPCRWAMRPL,ENSG00000188157.15:170_173:0:1053913:2-exons,1,ENSG00000188157.15,chr1,+,ref,1,1,nan,0,nan,nan,1053913;1053977;1054447;1054551;None;None,1053912;1053977;1054447;1054551;None;None,"(170, 173)",2-exonsJunction from above : 1053977;1054447

Laurie Prelot  20 days ago
foreground
gene_id  : 70
edge id in global file : 1170
edge id within gene: 173
segment id within gene : (168, 172)
segment corresponding coordinates: (168, 1053912, 1053977), (172, 1054447, 1054472),
(edited)

Laurie Prelot  20 days ago
background 
gene_id  : 70
edge id in global file : 3897
edge id within gene: 424
segment id within gene : (427, 435)
segment corresponding coordinates: (427, 1053976, 1053977), (435, 1054447, 1054454)


In [23]:
kmer= "SWHIGSRGK" #, "GCGSWHIGS"

In [25]:
eth_small.loc[eth_small['kmer'] == kmer, :]

kmer  isCrossJunction  TCGA-A2-A0D2-01A-21R-A034-07.all  \
5361  SWHIGSRGK             True                             180.0   

      TCGA-A2-A0SX-01A-12R-A084-07.all  TCGA-AO-A0JM-01A-21R-A056-07.all  \
5361                             397.0                             310.0   

      TCGA-BH-A18V-01A-11R-A12D-07.all  TCGA-C8-A12P-01A-11R-A115-07.all  
5361                             572.0                             154.0

In [41]:
# eth normalized

! for file_ in /cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/v2_v2.5f0752a_conf2_annotFrame_cap0_runs_pya0.17.1/TCGA_Breast_1102/filter_TCGA-*/*normalized*/*part*; \
do echo $file_; zcat ${file_} | grep SWHIGSRGK; done

/cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/v2_v2.5f0752a_conf2_annotFrame_cap0_runs_pya0.17.1/TCGA_Breast_1102/filter_TCGA-A2-A0D2-01A-21R-A034-07.all/TCGAA2A0D201A21RA03407all_generated_init_normalized/part-00000-cb15ca21-09a5-4238-8b32-868f0668274d-c000.csv.gz
SWHIGSRGK	246.16
/cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/v2_v2.5f0752a_conf2_annotFrame_cap0_runs_pya0.17.1/TCGA_Breast_1102/filter_TCGA-A2-A0SX-01A-12R-A084-07.all/TCGAA2A0SX01A12RA08407all_generated_init_normalized/part-00000-e4f50df0-3368-43de-a7d8-79208dabfa5f-c000.csv.gz
SWHIGSRGK	452.66
/cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/v2_v2.5f0752a_conf2_annotFrame_cap0_runs_pya0.17.1/TCGA_Breast_1102/filter_TCGA-AO-A0JM-01A-21R-A056-07.all/TCGAAOA0JM01A21RA05607all_generated_init_normalized/part-00000-279e5cac-54d6-43b8-bb77-3d234acdc200-c000.csv.gz
SWHIGSRGK	651.74
/cluster/work/grlab/projects/projects2020_OHSU/peptides_generation/v2_v2.5f0752a_conf2_a

In [43]:
ohsu.loc[: , [col for col in ohsu.columns if 'TCGA' not in col]]

jx  GTEx_brain_testis_over0  GTEx_brain_testis_over10  \
0  chr1;1053978;1054447;+                   1595.0                    1577.0   

   GTEx_brain_testis_over3  GTEx_breast_over0  GTEx_breast_over10  \
0                   1590.0              218.0               218.0   

   GTEx_breast_over3  GTEx_core_over0  GTEx_core_over10  GTEx_core_over3  ...  \
0              218.0           7799.0            7389.0           7685.0  ...   

   hanging_txs_included_inframe_pepseqs  prefiltered_in-frame_epitopes  \
0                                   NaN                            NaN   

   prefiltered_inframe_epitope_count  in-frame neoepitopes  \
0                                  0                   NaN   

   in-frame_neoepitope_count  \
0                          0   

               frame-agnostic_all-transcript_biexons  \
0  EKALQSNHFELSLRTEATQGLVLWSGKATERADYVALAIVDGHLQL...   

   nonhanging_prefiltered_epitopes  nonhanging_prefiltered_ep_count  \
0                              NaN                              0.0   

   nonhanging_neoepitopes nonhanging_neoep_count  
0                     NaN                    0.0  

[1 rows x 43 columns]

In [49]:
ohsu.loc[:, ['frame-agnostic_all-transcript_biexons']].values

array([['EKALQSNHFELSLRTEATQGLVLWSGKATERADYVALAIVDGHLQLSYNLGSQPVVLRSTVPVNTNRWLRVVAHREQREGSLQVGNEAPVTGSSPLGATQLDTDGALWLGGLPELPVGP']],
      dtype=object)

In [44]:
ohsu.loc[:, ['jx'] + target_samples_ohsu]

jx  TCGA-A2-A0D2-01A-21R-A034-07  \
0  chr1;1053978;1054447;+                    246.158054   

   TCGA-A2-A0SX-01A-12R-A084-07  TCGA-AO-A0JM-01A-21R-A056-07  \
0                    452.656363                    651.739725   

   TCGA-BH-A18V-01A-11R-A12D-07  TCGA-C8-A12P-01A-11R-A115-07  
0                    707.655859                    225.043383

In [26]:
ohsu.columns

Index(['jx', 'TCGA-24-1431-01A-01R-1566-13', 'TCGA-24-2298-01A-01R-1569-13',
       'TCGA-25-1313-01A-01R-1565-13', 'TCGA-25-1319-01A-01R-1565-13',
       'TCGA-61-2008-01A-02R-1568-13', 'TCGA-A2-A0D2-01A-21R-A034-07',
       'TCGA-A2-A0SX-01A-12R-A084-07', 'TCGA-AO-A0JM-01A-21R-A056-07',
       'TCGA-BH-A18V-01A-11R-A12D-07', 'TCGA-C8-A12P-01A-11R-A115-07',
       'GTEx_brain_testis_over0', 'GTEx_brain_testis_over10',
       'GTEx_brain_testis_over3', 'GTEx_breast_over0', 'GTEx_breast_over10',
       'GTEx_breast_over3', 'GTEx_core_over0', 'GTEx_core_over10',
       'GTEx_core_over3', 'GTEx_ovary_over0', 'GTEx_ovary_over10',
       'GTEx_ovary_over3', 'all_TCGA_over0', 'all_TCGA_over10',
       'all_TCGA_over3', 'BRCA_over_1', 'BRCA_over_0', 'OV_over_5',
       'OV_over_0', 'OV_over_1', 'BRCA_over_5', 'motif', 'annotation',
       '5';3'_coding_regions', 'has_cds', 'gene', 'gene_id', 'upstream_exon',
       'upstream_txs', 'downstream_exon', 'downstream_txs',
       'modified_upstream